### PR TITLE
chore(trie): remove `PrefixSetMut::contains`

### PR DESCRIPTION
**Pull request by rkrasiuk**
## Description

Remove `PrefixSetMut::contains` given that this is a mutable collection not intended for searching.

### CODE REVIEW

File: crates/trie/trie/src/prefix_set.rs [hunk 1]
```diff
@@ -75,65 +75,35 @@ pub struct TriePrefixSets {
...
+/// prefix_set_mut.insert(Nibbles::from_nibbles_unchecked(&[0xa, 0xb]));
+/// prefix_set_mut.insert(Nibbles::from_nibbles_unchecked(&[0xa, 0xb, 0xc]));
+/// let prefix_set = prefix_set_mut.freeze();
 /// assert!(prefix_set.contains(&[0xa, 0xb]));
 /// assert!(prefix_set.contains(&[0xa, 0xb, 0xc]));
 /// ```
 #[derive(Clone, Default, Debug)]
 pub struct PrefixSetMut {
     keys: Vec<Nibbles>,
-    sorted: bool,
```
Comment (rkrasiuk) [inline crates/trie/trie/src/prefix_set.rs:87]: should always be treated as unsorted

File: crates/trie/trie/benches/prefix_set.rs [hunk 2]
```diff
@@ -7,20 +7,36 @@ use proptest::{
...
     test_runner::{basic_result_cache, TestRunner},
 };
-use reth_trie::{prefix_set::PrefixSetMut, Nibbles};
+use reth_trie::{
+    prefix_set::{PrefixSet, PrefixSetMut},
+    Nibbles,
+};
 use std::collections::BTreeSet;
 
+pub trait PrefixSetMutAbstraction: Default {
```
Comment (mattsse) [inline crates/trie/trie/benches/prefix_set.rs:16]: needs docs

**Review by mattsse — CHANGES_REQUESTED** [review]
don't love the `Abstraction` suffix, be looks like we already use this this for PrefixSet

**Comment by rkrasiuk** [comment]
@mattsse doesn't matter really since this is only for benching

**Review by mattsse — APPROVED** [review]
gotcha

**Pull request merged**